### PR TITLE
Add details of review process to Contributor Handbook

### PR DESCRIPTION
**Pull request by spier**
We have used the new maturity levels for the patterns for over a year now. 
I figured we could aggregate our working knowledge in the Contributor Handbook, to make it easier for new contributors to understand what we are after when using the different maturity levels.

This PR also starts to implement #170 and #171.

## Changes made:

* Adds information about automatic checks 
* Describe some unspoken review expectations 
* Adds where the output artifacts in each level are published
* Document where to keep assets (images etc) that are used in patterns.
  * Most are kept in [/assets](https://github.com/InnerSourceCommons/InnerSourcePatterns/tree/master/assets), while the 3 [project-setup](https://github.com/InnerSourceCommons/InnerSourcePatterns/tree/master/patterns/2-structured/project-setup) patterns keep the assets in their own sub-folders
* (basic) first Style Guide and Glossary

### CODE REVIEW

**Comment by spier** [comment]
@maxcapraro as you wrote the first version of the Contributor Handbook, can you help us understand the intended semantic difference between the **Validation requirements** and **Content requirements**?

Understanding this better will help to make better decisions about which info to put where.

**Comment by spier** [comment]
@tapjdey @WillemJiang as you have made your first contributes to our Patterns recently, it would be great to get your feedback here.

This PR tries to explain better how we go about reviewing patterns.

You can focus your feedback on levels 1 and 2, as we don't have level 3 patterns yet.

If you have any further thoughts about this, let us know. And thanks for you help!

**Comment by spier** [comment]
@robtuley as you recently worked on a pattern, I was wondering if you could review if this PR here is a helpful improvement for the documentation in our Contributor Handbook?

File: meta/contributor-handbook.md [hunk 1]
```diff
@@ -18,42 +18,59 @@ For the first pattern you contribute, you should aim for maturity levels `1: Ini
...
 
 - Content requirements:
-  - Complies with the [patterns format](pattern-template.md)
-  - Complies with *basic style guide*(#) - *Oops! We have not yet developed this*
+  - Complies with the format described in the [Pattern  Template](pattern-template.md) - validated in parts by [Check: Pattern Syntax Validation](https://github.com/InnerSourceCommons/InnerSourcePatterns/actions?query=workflow%3A%22Pattern+Syntax+Validation%22)
+  - Complies with [Pattern Style Guide](pattern-style-guide.md)
+  - The title of the pattern is **memorable**, **concise**, and **descriptive** of what the pattern is about. For further tips see [Naming Patterns](pattern-style-guide.md#naming-patterns).
+  - Where appropriate this pattern links to other patterns of this level or higher.
+  - Links from the pattern to outside resources are working and are referencing a trusted resource - whether links are working is verified by [Check: Links](https://github.com/InnerSourceCommons/InnerSourcePatterns/actions?query=workflow%3A%22Link+Check+on+Patterns+and+README%22)
```
Comment (robtuley) [inline meta/contributor-handbook.md]: An alternative way to link to a specific workflow is via it's filepath which might be more stable than linking a search query for the workflow name e.g. the link checker runs are at https://github.com/InnerSourceCommons/InnerSourcePatterns/actions/workflows/link-checker.yml

**Review by robtuley — APPROVED** [review]
This looks like a helpful update to me as a recent contributor.

File: meta/contributor-handbook.md (hunk 1 above)
Comment (spier) [reply meta/contributor-handbook.md]: Ah, good one! Also makes for more readable links.
I fixed all instances of these. Thanks!

**Pull request merged**